pkg/segment/structs/evaluationstructs.go

```diff
@@ -814,8 +831,48 @@ func (self *TextExpr) EvaluateText(fieldToValue map[string]utils.CValueEnclosure
 			}
 		}
 		return minString, nil
-	}
 
+	} else if self.Op == "tostring" {
+		valueStr, err := self.Val.EvaluateToString(fieldToValue)
+		if err != nil {
+			return "", fmt.Errorf("TextExpr.Evaluate: failed to evaluate value for 'tostring' operation: %v", err)
+		}
+		if self.Format != nil {
+			formatStr, err := self.Format.Evaluate(fieldToValue)
+			if err != nil {
+				return "", fmt.Errorf("TextExpr.Evaluate: failed to evaluate format for 'tostring' operation: %v", err)
+			}
+			switch formatStr {
+			case "hex":
+				num, convErr := strconv.Atoi(valueStr)
+				if convErr != nil {
+					return "", fmt.Errorf("TextExpr.Evaluate: failed to convert value '%s' to integer for hex formatting: %v", valueStr, convErr)
+				}
+				return fmt.Sprintf("%#x", num), nil
+			case "commas":
+				num, convErr := strconv.ParseFloat(valueStr, 64)
+				if convErr != nil {
+					return "", fmt.Errorf("TextExpr.Evaluate: failed to convert value '%s' to float for comma formatting: %v", valueStr, convErr)
+				}
+				roundedNum := math.Round(num*100) / 100
+				formattedNum := humanize.CommafWithDigits(roundedNum, 2)
```

Comment:
Do you need `roundedNum := math.Round(num*100) / 100` since you're using `humanize.CommafWithDigits(roundedNum, 2)`? Does `CommafWithDigits()` not properly round the number how we want it?

Reply:
It was an oversight. Fixed it.

Reply:
If I use only humanize.CommafWithDigits(roundedNum, 2) it rounds off the number 12345.6789 to 12,345.67 but as per documentation the  number 12345.6789 is rounded to 12,345.68 that is why I used roundedNum := math.Round(num*100) / 100